packages/@lwc/engine/src/framework/hot-swaps.ts

```diff
@@ -0,0 +1,213 @@
+import { VM, scheduleRehydration } from './vm';
+import { isFalse, isUndefined, isNull } from '@lwc/shared';
+import { markComponentAsDirty, ComponentConstructor } from './component';
+import { Template } from './template';
+import { StylesheetFactory } from './stylesheet';
+
+const swappedTemplateMap: WeakMap<Template, Template> = new WeakMap();
+const swappedComponentMap: WeakMap<ComponentConstructor, ComponentConstructor> = new WeakMap();
+const swappedStyleMap: WeakMap<StylesheetFactory, StylesheetFactory> = new WeakMap();
```

Comment:
The type declaration can be avoided with the following shortcut.

```suggestion
const swappedTemplateMap = new WeakMap<Template, Template>();
const swappedComponentMap = new WeakMap<ComponentConstructor, ComponentConstructor>();
const swappedStyleMap = new WeakMap<StylesheetFactory, StylesheetFactory>();
```

Reply:
aren't both equivalent?